### DISCUSSION TITLE
Course Progress Tracker

### DISCUSSION THREAD

**Discussion by flamecopper** [facebook/docusaurus#10174] [General]
I am currently working on the following : https://docusaurus.io/

https://github.com/corey-richardson/course-progress-tracker

I'm thinking of implementing a course progress tracker.
So this is a node js application right.
How it goes is something like this:
Hi there,
I'm thinking of implementing a course progress tracker.
So this is a node js application right.
How it goes is something like this:

module1: (Based on total number of lines which results in the read time, This will aggregate all the topics below.)
	topic1: (15% of the total content)
	topic2: (20%)
	topic3: (20%)
	topic4: (15%)
	topic5: (20%)

On top of that, I'm thinking of implementing a login page for students to actually track their actual call total overall course progress.
Is there any plugin that can already do so? Or do I have to write my own code?, If I have to write my own code, which are the specific templates that I have to target?

**Comment by slorber** [comment]
I'm sorry but I don't understand the questions. 

Afaik there's no plugin for course template progression, and we are a static site generator so displaying any "user-specific data" can't be statically rendered and should rather be fetched/displayed at runtime in a modal or something.

> If I have to write my own code, which are the specific templates that I have to target?

You have to write code, and I can't advise which "template" (React theme components) without understanding the UX you want to create, preferably visually with mockups.

**Reply by flamecopper** [reply]
> I'm sorry but I don't understand the questions.
> 
> Afaik there's no plugin for course template progression, and we are a static site generator so displaying any "user-specific data" can't be statically rendered and should rather be fetched/displayed at runtime in a modal or something.
> 
> > If I have to write my own code, which are the specific templates that I have to target?
> 
> You have to write code, and I can't advise which "template" (React theme components) without understanding the UX you want to create, preferably visually with mockups.

Hi there. I can hope that I can make my questions a bit clearer.
So the component I'm talking about is DocCard and DocCategoryGeneratedIndexPage.
For example, this function in DocCard, how do I go to the doc and count the number of words?

function CardLink({item}) {


  const icon = isInternalUrl(item.href) ? '📄️' : '🔗';
  const doc = useDocById(item.docId ?? undefined);
 
  return (
    <CardLayout
      href={item.href}
      icon={icon}
      title={item.label}
      description={doc?.description}
    />
  );

  
}
